src/Compilers/VisualBasic/Portable/Binding/Binder_Symbols.vb

```diff
@@ -516,10 +511,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Next
 
                 If forwardedType IsNot Nothing Then
-                    If forwardedType.IsErrorType Then
-                        Dim diagInfo As DiagnosticInfo = DirectCast(forwardedType, ErrorTypeSymbol).ErrorInfo
-                        If diagInfo.Code = ERRID.ERR_TypeFwdCycle2 Then
-                            encounteredCycle = True
+                    If Not reportedAnError And forwardedType.IsErrorType Then
```

Comment:
BTW, from looking at the signature of the method, it is not clear that no diagnostics is going to be produced when reportedAnError is true. At the very least I think the parameter should be renamed to reportDiagnostics and its meaning should be reversed, but ideally we should remove it.